File: management/src/main/java/com/Airline/management/service/BookingService.java
package com.Airline.management.service;

import com.Airline.management.dto.BookingCountDto;
import com.Airline.management.model.Booking;

import java.util.List;

public interface BookingService {

    Booking saveBooking(Booking booking);

    List<Booking> getBookingsByPassenger(Long passengerId);
    Booking getBookingById(Long bookingId);
    Booking cancelBooking(Long bookingId);

    List<Booking> getUpcomingBookings(Long passengerId);
    List<Booking> getCancelledBookings(Long passengerId);
    List<Booking> getCompletedBookings(Long passengerId);
    BookingCountDto getBookingCounts(Long passengerId);
}

File: Frontend/airline_management_system/src/app/services/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Booking {
  bookingId: number;
  passengerId: number;
  flightId: number;
  travelClass: string;
  journeyDate: string;
  status: string;
  fare: number;
}

@Injectable({
  providedIn: 'root'
})
export class BookingService {

  private baseUrl = 'http://localhost:8080/api/bookings';

  constructor(private http: HttpClient) {}

  getBookingsByPassenger(
    passengerId: number
  ): Observable<Booking[]> {

    return this.http.get<Booking[]>(
      `${this.baseUrl}/passenger/${passengerId}`
    );
  }

  getUpcomingBookings(
    passengerId: number
  ): Observable<Booking[]> {

    return this.http.get<Booking[]>(
      `${this.baseUrl}/passenger/${passengerId}/upcoming`
    );
  }

  getCompletedBookings(
    passengerId: number
  ): Observable<Booking[]> {

    return this.http.get<Booking[]>(
      `${this.baseUrl}/passenger/${passengerId}/completed`
    );
  }

  getCancelledBookings(
    passengerId: number
  ): Observable<Booking[]> {

    return this.http.get<Booking[]>(
      `${this.baseUrl}/passenger/${passengerId}/cancelled`
    );
  }

  cancelBooking(
  bookingId: number
): Observable<Booking> {

  return this.http.put<Booking>(
    `${this.baseUrl}/${bookingId}/cancel`,
    {}
  );
} 
}